### PR TITLE
Source S3: Add start date filter for files

### PR DESCRIPTION
**Pull request by artem1205**
## What
Resolving https://github.com/airbytehq/airbyte/issues/7075

## How
Add `start_date` to filter files based on `lastmodified` option

## Recommended reading order

2. `y.python`

## 🚨 User Impact 🚨

no breaking changes

## Pre-merge Checklist

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Connector version has been incremented
    - [ ] Version has been bumped according to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines
    - [ ] `Dockerfile` has updated version
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` with an entry for the new version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)

- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

**Review by arsenlosenko — APPROVED** [review]

File: airbyte-integrations/connectors/source-s3/source_s3/stream.py [hunk 1]
```diff
@@ -53,10 +53,10 @@ def _list_bucket(self, accept_key: Callable = lambda k: True) -> Iterator[FileIn
             except KeyError:
                 pass
             else:
-                for c in content:
-                    key = c["Key"]
-                    if accept_key(key):
-                        yield FileInfo(key=key, last_modified=c["LastModified"], size=c["Size"])
+                for file in content:
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-s3/source_s3/stream.py:56]: As I remember we agreed move this functionality to `need_to_skip_file` method

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-s3/source_s3/stream.py (hunk 1 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-s3/source_s3/stream.py:56]: yes, but I've decided to leave it here, as `need_to_skip_file` used only for incremental sync, while we need this functionality for full refresh as well.

Comment (lazebnyi) [reply airbyte-integrations/connectors/source-s3/source_s3/stream.py:56]: Let's refactor this logic and add filtrating depend on sync type. 

 PCODE:
```
date = (stream_state or {}).get(cursor_field, start_date):
if last_mod > date: 
   process file 
```

Comment (artem1205) [reply airbyte-integrations/connectors/source-s3/source_s3/stream.py:56]: it seems to be not the optimal solution. 
Method `need_to_skip_file` is used only in incremental sync, and works with `stream_state` and `history` objects.

My goal is to modify the `list_objects` S3 method in such a way that the connector will basically only works with the filtered files.

Comment (artem1205) [reply airbyte-integrations/connectors/source-s3/source_s3/stream.py:62]: updated

**Review by lazebnyi — CHANGES_REQUESTED** [review]

**Comment by artem1205** [comment]
@lazebnyi ,

removed `need_to_skip_file` method, moved all filter logic in `filter_by_last_modified_date`

**Review by lazebnyi — APPROVED** [review]
Nice!

**Pull request merged**